src/Features/Core/Portable/PullMemberUp/MembersPuller.cs

```diff
@@ -162,17 +163,38 @@ private static ISymbol GetSymbolsToPullUp(MemberAnalysisResult analysisResult)
                 else
                 {
                     // We are pulling a public property, filter the non-public getter/setter.
-                    return CodeGenerationSymbolFactory.CreatePropertySymbol(
+                    member = CodeGenerationSymbolFactory.CreatePropertySymbol(
                         propertySymbol,
                         getMethod: FilterOutNonPublicAccessor(propertySymbol.GetMethod),
                         setMethod: FilterOutNonPublicAccessor(propertySymbol.SetMethod));
                 }
             }
-            else
+
+            // We don't support generating static interface members, so if we're coming from a static
+            // member, generating into an interface, then we need to update to non-static before generating.
+            if (member.IsStatic)
             {
-                // ICodeGenerationService will give the right result if it is method or event
-                return analysisResult.Member;
+                var modifier = DeclarationModifiers.From(member).WithIsStatic(false);
+                if (member is IMethodSymbol methodSymbol)
+                {
+                    return CodeGenerationSymbolFactory.CreateMethodSymbol(methodSymbol, modifiers: modifier);
+                }
+                else if (member is IPropertySymbol propSymbol)
```

Comment:
not sure if I like the flow here where property symbol ends up being checked up top, then modified down here.

I wonder if its possible to invert a little bit - basically build the modifiers with is static or not once -then switch/condition off of the kind of symbol and return the result directly.

Reply:
Will give it a go. I was trying to avoid recreating the symbols unless necessary, just in case, but I guess if no tests break then no harm done :)